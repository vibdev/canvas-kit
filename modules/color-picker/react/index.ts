import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
