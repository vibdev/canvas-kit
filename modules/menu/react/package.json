{
  "name": "@workday/canvas-kit-react-menu",
  "version": "0.0.24",
  "description": "A container for navigation or action items",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "menu"
  ],
  "devDependencies": {
    "@material-ui/core": "^3.9.2",
    "@types/react-transition-group": "^2.0.14",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "react-transition-group": "^2.5.0"
  },
  "dependencies": {
    "@types/uuid": "^3.4.4",
    "@workday/canvas-kit-react-card": "^0.3.23",
    "@workday/canvas-kit-react-common": "^0.7.5",
    "@workday/canvas-kit-react-core": "^1.10.6",
    "@workday/canvas-kit-react-icon": "^1.7.7",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12",
    "uuid": "^3.3.2"
  },
  "peerDependencies": {
    "react": ">= 16.3 < 17"
  }
}
