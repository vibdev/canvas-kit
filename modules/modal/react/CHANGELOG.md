# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/modal/compare/@workday/canvas-kit-react-modal@0.0.1...@workday/canvas-kit-react-modal@0.0.2) (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal





## 0.0.1 (2019-06-20)

**Note:** Version bump only for package @workday/canvas-kit-react-modal
