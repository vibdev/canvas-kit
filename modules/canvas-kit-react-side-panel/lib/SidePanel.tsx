import * as React from 'react';
import styled from 'react-emotion';
import {colors, spacing, type, CanvasSpacingValue} from '@workday/canvas-kit-react-core';
import {IconButton, ButtonSizes} from '@workday/canvas-kit-react-button';
import {arrowLeftIcon, arrowRightIcon} from '@workday/canvas-system-icons-web';

export interface SidePanelProps {
  open: boolean;
  onToggleClick?: () => void;
  title?: string | React.ReactNode;
  openRight?: boolean;
  openLeft?: boolean;
  padding?: CanvasSpacingValue;
}

const openSidePanelWidth = 300;

const Title = styled('h2')({
  ...type.h2,
  marginTop: spacing.zero,
});

const SidePanelToggleContainer = styled('div')({
  position: 'absolute',
  bottom: spacing.s,
});

const SidePanelContainer = styled('div')<SidePanelProps>(
  {
    height: '100%',
    boxSizing: 'border-box',
    display: 'flex',
    flexDirection: 'column',
    transition: 'width 150ms ease-out 0s',
    position: 'absolute',
  },
  ({open, openRight, openLeft, padding}) => ({
    boxShadow: !open ? '0 8px 16px -8px rgba(0, 0, 0, 0.16)' : '',
    backgroundColor: open ? colors.soap100 : colors.frenchVanilla100,
    width: open ? openSidePanelWidth : spacing.xl,
    padding: open ? padding || spacing.m : spacing.zero,
    right: openRight && !openLeft ? spacing.zero : '',
    left: openLeft && !openRight ? spacing.zero : '',
    alignItems: open ? '' : 'center',
    [`${SidePanelToggleContainer}`]: {
      right: open && !openRight ? spacing.s : '',
      left: openRight && open ? spacing.s : '',
    },
  })
);

export default class MyComponent extends React.Component<SidePanelProps> {
  public render() {
    const {title, onToggleClick, open, openRight, openLeft, padding} = this.props;

    let toggleButtonDirection;
    if (!openRight) {
      toggleButtonDirection = open ? arrowLeftIcon : arrowRightIcon;
    } else {
      toggleButtonDirection = open ? arrowRightIcon : arrowLeftIcon;
    }
    return (
      <SidePanelContainer
        aria-expanded={open}
        aria-orientation="vertical"
        padding={padding}
        openLeft={openLeft}
        openRight={openRight}
        open={open}
      >
        {title && open ? <Title>{title}</Title> : null}
        {this.props.children}
        {onToggleClick && (
          <SidePanelToggleContainer>
            <IconButton
              buttonSize={ButtonSizes.Small}
              onClick={onToggleClick}
              icon={toggleButtonDirection}
              buttonType={IconButton.Types.Filled}
            />
          </SidePanelToggleContainer>
        )}
      </SidePanelContainer>
    );
  }
}
