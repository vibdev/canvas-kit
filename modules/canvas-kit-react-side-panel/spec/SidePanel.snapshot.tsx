import * as React from 'react';
import SidePanel from '../lib/SidePanel';
import * as renderer from 'react-test-renderer';

describe('SidePanel Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<SidePanel open={true} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a closed side panel', () => {
    const component = renderer.create(<SidePanel open={false} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a side panel header', () => {
    const component = renderer.create(<SidePanel title={'Side Panel Header'} open={true} />);
    expect(component).toMatchSnapshot();
  });
  test('does not render header when side panel is closed', () => {
    const component = renderer.create(<SidePanel title={'Side Panel Header'} open={false} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a toggle button', () => {
    const component = renderer.create(
      <SidePanel onToggleClick={jest.fn} title={'Side Panel Header'} open={false} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders side panel on the right with correct toggle button when panel is closed', () => {
    const component = renderer.create(
      <SidePanel
        onToggleClick={jest.fn}
        openRight={true}
        title={'Side Panel Header'}
        open={false}
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders side panel on the right with correct toggle button when panel is open', () => {
    const component = renderer.create(
      <SidePanel onToggleClick={jest.fn} openRight={true} title={'Side Panel Header'} open={true} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders side panel on the left', () => {
    const component = renderer.create(
      <SidePanel openLeft={true} title={'Side Panel Header'} open={true} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a side panel with children elements', () => {
    const component = renderer.create(
      <SidePanel open={true}>
        <ul>
          <li>Home</li>
          <li>Favorites</li>
          <li>Recents</li>
          <li>Trash</li>
        </ul>
      </SidePanel>
    );
    expect(component).toMatchSnapshot();
  });
});
