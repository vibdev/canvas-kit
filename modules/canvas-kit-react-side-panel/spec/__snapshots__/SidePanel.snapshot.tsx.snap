// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel Snapshots does not render header when side panel is closed 1`] = `
.emotion-0 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 8px 16px -8px rgba(0,0,0,0.16);
  background-color: #ffffff;
  width: 64px;
  padding: 16px 0;
}

.emotion-0 .ezxdcmp1 {
  right: px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-0 emotion-1"
  onBreakpointChange={[Function]}
  open={false}
  role="region"
/>
`;

exports[`SidePanel Snapshots renders a closed side panel 1`] = `
.emotion-0 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 8px 16px -8px rgba(0,0,0,0.16);
  background-color: #ffffff;
  width: 64px;
  padding: 16px 0;
}

.emotion-0 .ezxdcmp1 {
  right: px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-0 emotion-1"
  onBreakpointChange={[Function]}
  open={false}
  role="region"
/>
`;

exports[`SidePanel Snapshots renders a side panel from the right 1`] = `
.emotion-0 {
  font-size: 24px;
  line-height: 32px;
  font-weight: 700;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-top: 0px;
}

.emotion-2 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: 0px;
  left: px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-2 .ezxdcmp1 {
  right: px;
  left: 16px;
}

<div
  aria-orientation="vertical"
  className="emotion-2 emotion-3"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
>
  <h2
    className="emotion-0 emotion-1"
  >
    Side Panel Header
  </h2>
</div>
`;

exports[`SidePanel Snapshots renders a side panel header 1`] = `
.emotion-2 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-2 .ezxdcmp1 {
  right: 16px;
  left: px;
}

.emotion-0 {
  font-size: 24px;
  line-height: 32px;
  font-weight: 700;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-top: 0px;
}

<div
  aria-orientation="vertical"
  className="emotion-2 emotion-3"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
>
  <h2
    className="emotion-0 emotion-1"
  >
    Side Panel Header
  </h2>
</div>
`;

exports[`SidePanel Snapshots renders a side panel with children elements 1`] = `
.emotion-0 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-0 .ezxdcmp1 {
  right: 16px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-0 emotion-1"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
>
  <ul>
    <li>
      Home
    </li>
    <li>
      Favorites
    </li>
    <li>
      Recents
    </li>
    <li>
      Trash
    </li>
  </ul>
</div>
`;

exports[`SidePanel Snapshots renders a toggle button 1`] = `
.emotion-4 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 8px 16px -8px rgba(0,0,0,0.16);
  background-color: #ffffff;
  width: 64px;
  padding: 16px 0;
}

.emotion-4 .emotion-1 {
  right: px;
  left: px;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 32px;
  height: 32px;
  position: absolute;
  bottom: 16px;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-2 span .wd-icon-fill,
.emotion-2 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-2:focus span .wd-icon-fill,
.emotion-2:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover:focus {
  background-color: #dfe2e6;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ced3d9;
}

.emotion-2:hover {
  background-color: #dfe2e6;
}

.emotion-2:hover span .wd-icon-fill,
.emotion-2:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-2:disabled span .wd-icon-fill,
.emotion-2:active:disabled span .wd-icon-fill,
.emotion-2:focus:disabled span .wd-icon-fill,
.emotion-2:hover:disabled span .wd-icon-fill,
.emotion-2:disabled span .wd-icon-accent,
.emotion-2:active:disabled span .wd-icon-accent,
.emotion-2:focus:disabled span .wd-icon-accent,
.emotion-2:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2 span svg {
  width: 20px;
  height: 20px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<div
  aria-orientation="vertical"
  className="emotion-4 emotion-5"
  onBreakpointChange={[Function]}
  open={false}
  role="region"
>
  <button
    aria-label="show navigation"
    aria-pressed={false}
    className="emotion-1 emotion-2 emotion-3"
    onClick={[Function]}
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-arrow-right wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M10.525 5.263a.506.506 0 0 1 .003-.71l.7-.701a.495.495 0 0 1 .711-.003l7.778 7.777a.498.498 0 0 1 .144.352.48.48 0 0 1-.144.356l-7.778 7.778a.506.506 0 0 1-.71-.003l-.701-.7a.495.495 0 0 1-.003-.711l5.702-5.703H4.6a.505.505 0 0 1-.5-.505v-.99c0-.29.224-.505.5-.505h11.657l-5.732-5.732z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
</div>
`;

exports[`SidePanel Snapshots renders as expected 1`] = `
.emotion-0 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-0 .ezxdcmp1 {
  right: 16px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-0 emotion-1"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
/>
`;

exports[`SidePanel Snapshots renders side panel on the left 1`] = `
.emotion-0 {
  font-size: 24px;
  line-height: 32px;
  font-weight: 700;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-top: 0px;
}

.emotion-2 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: px;
  left: 0px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-2 .ezxdcmp1 {
  right: 16px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-2 emotion-3"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
>
  <h2
    className="emotion-0 emotion-1"
  >
    Side Panel Header
  </h2>
</div>
`;

exports[`SidePanel Snapshots renders side panel on the right with correct toggle button when panel is closed 1`] = `
.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 32px;
  height: 32px;
  position: absolute;
  bottom: 16px;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-2 span .wd-icon-fill,
.emotion-2 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-2:focus span .wd-icon-fill,
.emotion-2:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover:focus {
  background-color: #dfe2e6;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ced3d9;
}

.emotion-2:hover {
  background-color: #dfe2e6;
}

.emotion-2:hover span .wd-icon-fill,
.emotion-2:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-2:disabled span .wd-icon-fill,
.emotion-2:active:disabled span .wd-icon-fill,
.emotion-2:focus:disabled span .wd-icon-fill,
.emotion-2:hover:disabled span .wd-icon-fill,
.emotion-2:disabled span .wd-icon-accent,
.emotion-2:active:disabled span .wd-icon-accent,
.emotion-2:focus:disabled span .wd-icon-accent,
.emotion-2:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-2 span svg {
  width: 20px;
  height: 20px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-4 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: 0px;
  left: px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 8px 16px -8px rgba(0,0,0,0.16);
  background-color: #ffffff;
  width: 64px;
  padding: 16px 0;
}

.emotion-4 .emotion-1 {
  right: px;
  left: px;
}

<div
  aria-orientation="vertical"
  className="emotion-4 emotion-5"
  onBreakpointChange={[Function]}
  open={false}
  role="region"
>
  <button
    aria-label="show navigation"
    aria-pressed={false}
    className="emotion-1 emotion-2 emotion-3"
    onClick={[Function]}
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-arrow-left wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M13.436 5.263a.506.506 0 0 0-.003-.71l-.7-.701a.495.495 0 0 0-.71-.003l-7.779 7.777a.498.498 0 0 0-.144.352.48.48 0 0 0 .144.356l7.778 7.778a.506.506 0 0 0 .71-.003l.701-.7a.495.495 0 0 0 .003-.711l-5.702-5.703H19.36c.273 0 .5-.226.5-.505v-.99a.495.495 0 0 0-.5-.505H7.704l5.732-5.732z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
</div>
`;

exports[`SidePanel Snapshots renders side panel on the right with correct toggle button when panel is open 1`] = `
.emotion-0 {
  font-size: 24px;
  line-height: 32px;
  font-weight: 700;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-top: 0px;
}

.emotion-6 {
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: width 150ms ease-out 0s;
  transition: width 150ms ease-out 0s;
  position: absolute;
  right: 0px;
  left: px;
  -webkit-align-items: px;
  -webkit-box-align: px;
  -ms-flex-align: px;
  align-items: px;
  box-shadow: px;
  background-color: #f6f7f8;
  width: 300px;
  padding: 24px;
}

.emotion-6 .emotion-3 {
  right: px;
  left: 16px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: #f0f1f2;
  width: 32px;
  height: 32px;
  position: absolute;
  bottom: 16px;
}

.emotion-4:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-4:disabled,
.emotion-4:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-4 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-4 span .wd-icon-fill,
.emotion-4 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-4:focus span .wd-icon-fill,
.emotion-4:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-4:hover:focus {
  background-color: #dfe2e6;
}

.emotion-4:active,
.emotion-4:focus:active,
.emotion-4:hover:active {
  background-color: #ced3d9;
}

.emotion-4:hover {
  background-color: #dfe2e6;
}

.emotion-4:hover span .wd-icon-fill,
.emotion-4:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-4:disabled,
.emotion-4:active:disabled,
.emotion-4:focus:disabled,
.emotion-4:hover:disabled {
  background-color: #f6f7f8;
}

.emotion-4:disabled span .wd-icon-fill,
.emotion-4:active:disabled span .wd-icon-fill,
.emotion-4:focus:disabled span .wd-icon-fill,
.emotion-4:hover:disabled span .wd-icon-fill,
.emotion-4:disabled span .wd-icon-accent,
.emotion-4:active:disabled span .wd-icon-accent,
.emotion-4:focus:disabled span .wd-icon-accent,
.emotion-4:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-4:not([disabled]):focus,
.emotion-4:not([disabled]):active {
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-4 span svg {
  width: 20px;
  height: 20px;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

<div
  aria-orientation="vertical"
  className="emotion-6 emotion-7"
  onBreakpointChange={[Function]}
  open={true}
  role="region"
>
  <h2
    className="emotion-0 emotion-1"
  >
    Side Panel Header
  </h2>
  <button
    aria-label="hide navigation"
    aria-pressed={false}
    className="emotion-3 emotion-4 emotion-5"
    onClick={[Function]}
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-arrow-right wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M10.525 5.263a.506.506 0 0 1 .003-.71l.7-.701a.495.495 0 0 1 .711-.003l7.778 7.777a.498.498 0 0 1 .144.352.48.48 0 0 1-.144.356l-7.778 7.778a.506.506 0 0 1-.71-.003l-.701-.7a.495.495 0 0 1-.003-.711l5.702-5.703H4.6a.505.505 0 0 1-.5-.505v-.99c0-.29.224-.505.5-.505h11.657l-5.732-5.732z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </button>
</div>
`;
